src/rust/ensogl/src/math/geometry/triangle.rs

```diff
@@ -0,0 +1,128 @@
+//! This module defines an computations related to triangle geometry.
+
+use crate::math::algebra::{Acos,Sin,Cos,Sqrt};
+use crate::math::algebra::Field;
+
+use core::f32::consts::PI;
+
+/// Represents a triangle through its angles and side lengths. This struct is only meant to be a
+/// return type within this module and should not otherwise be instantiated.
+///
+///
+/// Triangle Schematic
+/// ----------------------
+///
+///                           C
+///                          / \
+///                        /    -\
+///                      -/ gamma --\
+///                   --/            --\
+///               b -/                  --\ a
+///              --/                       --\
+///            -/                             --\
+///          -/                                  --\
+///       --/                                       --\
+///     -/  alpha                                   beta-\
+///   -/--------------------------------------------------- B
+///  A                              c
```

Comment:
Merry Christmas! :D

Reply:
What do you think about changing the variables and the picture to something simpler?

```
///   C
///  / \
/// A---B
```

Plus changing names to `side_ab` (the distance between `A` and `B`), `side_bc`, `side_ac`, `angle_a`, `angle_b`, and `angle_c` ?

Reply:
It's shorter so I'll go for it.
I cleaned up the diagram, but I'd like to keep the side and angle descriptions in there for reference.